lib/src/protocol/proxy_protocol/send.rs

```diff
@@ -54,31 +53,39 @@ impl <Front:SocketHandler + Read> SendProxyProtocol<Front> {
     debug!("Writing proxy protocol header");
 
     if let Some(ref mut socket) = self.backend {
-      if let Some(ref mut header) = self.header {
-        loop {
-          match socket.write(&mut header[self.cursor_header..]) {
-            Ok(sz) => {
-              self.cursor_header += sz;
-              metrics.backend_bout += sz;
-
-              if self.cursor_header == header.len() {
-                debug!("Proxy protocol sent, upgrading");
-                return (ProtocolResult::Upgrade, SessionResult::Continue);
-              }
-            },
-            Err(e) => match e.kind() {
-              ErrorKind::WouldBlock => {
-                self.back_readiness.event.remove(Ready::writable());
-                return (ProtocolResult::Continue, SessionResult::Continue);
+      match gen_proxy_protocol_header(&self.frontend, socket) {
```

Comment:
if we do it this, way, the header will be generated every time we go in `back_writable`.
Instead, it should check if `self.header.is_none()`, generate and store it if it is, then go to the writing phase